src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -593,74 +596,175 @@ const EnvironmentPicker = (props: EnvironmentPickerProps) => {
-                hidden={isDialogHidden}
+                hidden={false}
+                styles={{
+                    root: { display: !isDialogHidden ? 'flex' : 'none' }
+                }}
                 onDismiss={handleOnDismiss}
                 dialogContentProps={dialogContentProps}
                 modalProps={modalProps}
             >
-                <div className="cb-environment-picker-dialog-form">
-                    <ComboBox
-                        placeholder={t('environmentPicker.enterEnvironmentUrl')}
-                        label={t('environmentPicker.environmentUrl')}
-                        allowFreeform={true}
-                        autoComplete={'on'}
-                        options={environmentOptions}
-                        styles={comboBoxStyles}
-                        required
-                        text={getUrl(environmentToEdit)}
-                        onChange={(_e, option, _idx, value) =>
-                            handleOnEnvironmentUrlChange(option, value)
-                        }
-                        errorMessage={environmentInputError}
-                        onRenderOption={(option) =>
-                            onRenderOption(option, 'environment')
-                        }
-                        onRenderLabel={(p) => (
-                            <div className="cb-environment-picker-environment-url-label">
-                                <span className="cb-environment-picker-environment-url-label-text">
-                                    {p.props.label}
-                                </span>
-                                {environmentsState.isLoading && (
-                                    <Spinner
-                                        size={SpinnerSize.xSmall}
-                                        label={t('loadingInstances')}
-                                        ariaLive="assertive"
-                                        labelPosition="right"
-                                    />
-                                )}
-                            </div>
-                        )}
-                        selectedKey={getUrl(environmentToEdit)}
-                    />
-                    {props.storage && (
-                        <ComboBox
-                            placeholder={t(
-                                'environmentPicker.enterContainerUrl'
-                            )}
-                            label={t('environmentPicker.containerUrl')}
-                            allowFreeform={true}
-                            autoComplete={'on'}
-                            options={containerOptions}
-                            styles={comboBoxStyles}
-                            required
-                            text={containerUrlToEdit}
-                            onChange={(_e, option, _idx, value) =>
-                                handleOnContainerUrlChange(option, value)
+                {firstTimeVisible && (
```

Comment:
firstTimeVisible is used not to render resource picker components with data fetch requests if the dialog has not opened yet for the first time. We want to trigger the adapter data fetching when we open the dialog for the first time and keep the same component mounted and same data request flow going as we close and open the dialog afterwards.

The issue is when Dialog fluent component is open and closed the content of it [is being unmounted](https://github.com/microsoft/fluentui/issues/11529) (all the ResourcePicker children components) - so it was a problem since we dont want to unmount and mount and refetch the resources again and want to keep the same loading process going while we close and open the dialog by clicking edit icon. That is why I applied the suggested solution for it - just always visible but changing the css. This also answers your other question below.